pubsub.go

```diff
@@ -137,27 +187,43 @@ func (p *PubsubValueStore) Subscribe(key string) error {
 	}
 
 	p.mx.Lock()
-	existingSub, bootstraped := p.subs[key]
+	existingSub, _ := p.subs[key]
 	if existingSub != nil {
 		p.mx.Unlock()
 		sub.Cancel()
 		return nil
 	}
 
 	p.subs[key] = sub
-	ctx, cancel := context.WithCancel(p.ctx)
-	go p.handleSubscription(sub, key, cancel)
+	go p.handleSubscription(sub, key)
 	p.mx.Unlock()
 
+	go p.rebroadcast(key)
 	log.Debugf("PubsubResolve: subscribed to %s", key)
 
-	if !bootstraped {
-		// TODO: Deal with publish then resolve case? Cancel behaviour changes.
-		go bootstrapPubsub(ctx, p.cr, p.host, topic)
-	}
 	return nil
 }
 
+func (p *PubsubValueStore) rebroadcast(key string) {
+	topic := KeyToTopic(key)
+	time.Sleep(p.rebroadcastInitialDelay)
+
+	ticker := time.NewTicker(p.rebroadcastInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
```

Comment:
Makes sense. We'll need to do another goroutine encapsulation because `pubsub.Publish` has no context.

I think this was a holdover from before when Publish and Subscribe were independent on the `PubSubValueStore`, but now that Publish implies Subscribe canceling Subscribe should probably cancel Publish.